spa/src/components/content/contributor-portal/ContributorPortal.tsx

```diff
@@ -0,0 +1,189 @@
+import CheckIcon from '@material-design-icons/svg/outlined/check.svg?react';
+import SaveIcon from '@material-design-icons/svg/outlined/save.svg?react';
+import { AxiosError } from 'axios';
+import { Button, TextField } from 'components/base';
+import Hero from 'components/common/Hero';
+import GenericErrorBoundary from 'components/errors/GenericErrorBoundary';
+import { GENERIC_ERROR } from 'constants/textConstants';
+import { RevenueProgram } from 'hooks/useContributionPage';
+import { useRevenueProgram } from 'hooks/useRevenueProgram';
+import PropTypes, { InferProps } from 'prop-types';
+import { useEffect, useMemo, useState } from 'react';
+import { useAlert } from 'react-alert';
+import { Controller, useForm } from 'react-hook-form';
+import { SectionWrapper } from '../pages/Pages.styled';
+import {
+  ActionWrapper,
+  Description,
+  FormWrapper,
+  InputsWrapper,
+  Label,
+  SuccessMessage
+} from './ContributorPortal.styles';
+
+type ContactInfoFormFields = {
+  contact_email: string;
+  contact_phone: string;
+};
+
+export interface ContributorPortalProps extends InferProps<typeof ContributorPortalPropTypes> {
+  revenueProgram?: RevenueProgram;
+}
+
+const ContributorPortal = ({ revenueProgram }: ContributorPortalProps) => {
+  const alert = useAlert();
+  const [showSuccess, setShowSuccess] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<Record<keyof ContactInfoFormFields, string>>();
+  const { updateRevenueProgram } = useRevenueProgram(revenueProgram?.id);
+
+  const {
+    control,
+    watch,
+    reset,
+    handleSubmit,
+    formState: { errors }
+  } = useForm<ContactInfoFormFields>({
+    defaultValues: {
+      contact_email: revenueProgram?.contact_email ?? '',
+      contact_phone: revenueProgram?.contact_phone ?? ''
+    }
+  });
+
+  const contact_email = watch('contact_email');
+  const contact_phone = watch('contact_phone');
+
+  useEffect(() => {
+    setShowSuccess(false);
+  }, [contact_email, contact_phone, setShowSuccess]);
+
+  const isDifferent = useMemo(
+    () => ({
+      contact_email: contact_email !== (revenueProgram?.contact_email ?? ''),
+      contact_phone: contact_phone !== revenueProgram?.contact_phone ?? ''
+    }),
+    [contact_email, contact_phone, revenueProgram?.contact_email, revenueProgram?.contact_phone]
+  );
```

Comment:
```suggestion
  const isDifferent = useMemo(
    () => contact_email !== (revenueProgram?.contact_email ?? '') || contact_phone !== (revenueProgram?.contact_phone ?? ''),
    [contact_email, contact_phone, revenueProgram?.contact_email, revenueProgram?.contact_phone]
  );
```
Since it looks like you don't need to distinguish between changes on each field, only that a field was changed.